url7a45f192ca_mfullen_ping_pong_net_tgz-pJ8-ping_pong_net_client_io_TestClientNonPPNServerJ8/src/ping/pong/net/server/ClientExample.java

```diff
@@ -0,0 +1,302 @@
+        }
+    }
+
+    public static void ssl1() throws IOException,
+                                     InterruptedException,
+                                     NoSuchAlgorithmException,
+                                     KeyManagementException,
+                                     KeyStoreException,
+                                     UnrecoverableKeyException,
+                                     CertificateException,
+                                     URISyntaxException
+    {
+        {
+
+//            KeyStore ks = KeyStore.getInstance("JKS");
+//
+//            char[] pw = "pingpong123".toCharArray();
+//            ks.load(Thread.currentThread().getContextClassLoader().getResourceAsStream(ConnectionConfiguration.DEFAULT_KEY_STORE), pw);
+//            KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("SunX509");
+//            keyManagerFactory.init(ks, pw);
+//            SSLContext ctx = SSLContext.getInstance("SSLv3");
+//            ctx.init(keyManagerFactory.getKeyManagers(), null, new SecureRandom());
+////            ctx.init(new KeyManager[0], new TrustManager[]
+////                    {
+////                        new DefaultTrustManager()
+////                    }, new SecureRandom());
+//            SSLContext.setDefault(ctx);
+            // SSLSocketFactory factory = (SSLSocketFactory) ctx.getSocketFactory();
+
+
+            SSLContext ctx = SSLContext.getInstance("SSLv3");
+            ctx.init(null, null, null);
+
+
+            String absoluteKeyPath = new File(Thread.currentThread().getContextClassLoader().getResource(ConnectionConfiguration.DEFAULT_KEY_STORE).toURI()).getAbsolutePath();
+            System.out.println(absoluteKeyPath);
+            System.setProperty("javax.net.ssl.keyStore", absoluteKeyPath);
+            System.setProperty("javax.net.ssl.keyStorePassword", ConnectionConfiguration.DEFAULT_KEY_STORE_PASSWORD);
+            System.setProperty("java.protocol.handler.pkgs", "com.sun.net.ssl.internal.www.protocol");
+
+            String absolutePath = new File(Thread.currentThread().getContextClassLoader().getResource("pingpong.crt").toURI()).getAbsolutePath();
+            System.out.println(absolutePath);
+            System.setProperty("javax.net.ssl.trustStore", absoluteKeyPath);
+            System.setProperty("javax.net.ssl.trustStorePassword", "pingpong123");
+
+            // SSLSocketFactory factory = (SSLSocketFactory) ctx.getSocketFactory();
+            SSLSocketFactory factory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+            // SSLSocket client = (SSLSocket) factory.createSocket("localhost", 5011);
+            Socket client = (SSLSocket) factory.createSocket("localhost", 5011);
+            //  client.startHandshake();
+            // java.security.cert.Certificate[] serverCerts = client.getSession().getPeerCertificates();
+            ObjectInputStream inputStream = new ObjectInputStream(client.getInputStream());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *inputStream*. The resource is referenced at lines 91, 94, 95, 97, 98, 102. The resource is closed at line 103. There are other execution paths that do not contain closure statements, for example, when ObjectOutputStream.flush throws an exception. Ensure *inputStream* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).